driver/ni/ni.h

```diff
@@ -25,448 +25,434 @@
-                                const breaker::Config breaker_config); 
-
-        void exec(task::Command &cmd) override;
-        void stop() override;
-        void start();
-
-        static std::unique_ptr<task::Task> configure(   const std::shared_ptr<task::Context> &ctx,
-                                                        const synnax::Task &task);
-        bool ok();
-        ~WriterTask(){
-            LOG(INFO) << "WriterTask destructor called";
-        }
-
-
-    private:
-        std::atomic<bool>  running = false;
-        std::shared_ptr<task::Context> ctx;
-        synnax::Task task;
-        pipeline::Control cmd_write_pipe;
-        pipeline::Acquisition state_write_pipe;
-        bool ok_state = true;
-    };
-
-
-    ///////////////////////////////////////////////////////////////////////////////////
-    //                                    Factory                                    //
-    ///////////////////////////////////////////////////////////////////////////////////
-    class Factory final : public task::Factory{
-    public:
-        std::pair<std::unique_ptr<task::Task>, bool> configureTask( const std::shared_ptr<task::Context> &ctx,
-                                                                    const synnax::Task &task) override;
-
-        std::vector<std::pair<synnax::Task, std::unique_ptr<task::Task> > >
-        configureInitialTasks(      const std::shared_ptr<task::Context> &ctx,
-                                    const synnax::Rack &rack) override;
-
-        ~Factory() = default;
-
-    };
-}
+typedef struct ReaderConfig {
+    std::string device_key;
+    std::vector<ChannelConfig> channels;
+    std::uint64_t sample_rate = 0;
+    std::uint64_t stream_rate = 0;
+    std::string device_name;
+    std::string task_name;
+    std::string timing_source; // for sample clock
+    std::uint64_t period = 0;
+    synnax::ChannelKey task_key;
+    std::set<uint32_t> index_keys;
+} ReaderConfig;
 
+/////////////////////////////////////////////////////////////////////////////////////////////
+//                                                                                         //
+//                                     DAQ INTERFACES                                      //
+//                                                                                         //
+/////////////////////////////////////////////////////////////////////////////////////////////
 
+///////////////////////////////////////////////////////////////////////////////////
+//                                    NiSource                                   //
+///////////////////////////////////////////////////////////////////////////////////
+class Source : public pipeline::Source {
+public:
+    explicit Source(TaskHandle task_handle,
+                    const std::shared_ptr<task::Context> &ctx,
+                    const synnax::Task task);
 
 
-/*
+    int init();
```

Comment:
Why do we need an `init` method? Can't we just do this in the constructor? Or make this a static method that also constructs the source?